pkg/metrics/otelmetrics/otelCounter.go

```diff
@@ -0,0 +1,37 @@
+package otelmetrics
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+type otelCounter struct {
+	counter      metric.Int64Counter
+	attributeSet attribute.Set
+}
+
+func newOtelCounter(name string, tags map[string]string) *otelCounter {
+	meter := otel.Meter("jaeger-V2")
+	counter, err := meter.Int64Counter(name)
+	if err != nil {
```

Comment:
What could cause an error here? We never panic in library functions.

Reply:
It returns an error as well, I was following the initialization of Counter as mentioned in https://opentelemetry.io/docs/languages/go/instrumentation/

Reply:
I'm asking why it returns an error. Under what conditions?

Reply:
The documentation dose not specify the conditions for error, but I believe if you fail to set a meterProvider initially, the otel.Meter call will return a no-op meter which will fail to generate any data, and initializing a counter using that meter may raise an error.

Reply:
We should look in the code directly, to understand what may cause an error. Ideally our API would've returned error as well, but it's a massive change. In the past we had implementations that would return an error if the same metric was attempted to be created twice (which sometimes happens in our code), so we had to put a cache in front of that implementation. Another reason for errors (e.g. in Prometheus) is when the strings are not acceptable to the specific backend, in which case we sanitize them first.

Basically, Jaeger creates a lot of metrics at runtime, e.g. for counting different services & operations observed in spans. So we absolutely cannot panic at runtime, especially for errors that are known upfront and can be alleviated.

Reply:
Yes makes total sense, errors at runtime would be disastrous. So, primary task is to look into the underlying implementation of Counter in OTEL's code and understand all the possible reasons that it would raise an error?

Reply:
On it!